Documentation/proposal-runconfig_inside_osv.md

```diff
@@ -0,0 +1,40 @@
+# Proposal for meta/run.yaml content being present in the composed image
+
+Currently, Capstan runtime support makes use of `$CWD/meta/run.yaml` for composing and running an OSv image. Although this is fine for the composing, it gets tricky for running because `capstan run <IMAGE>` generally does not require user to be inside package root directory.
+
+I propose the following solution:
+
+### Composing image
+File meta/run.yaml is present at package root directory (same as until now)
+
+When user runs `capstan package compose` the content of meta/run.yaml file is processed and new files are generated inside the OSv image folder `/osv-runtime`:
+
+Capstan should create a separate file for each named configuration e.g.
+```
+/osv-runtime
+   |--burek
+   |--pizza   <--------\
+   |                    |
+   |--default-runtime  /
+```
+Each runtime file should contain only one line representing the run command that is to be run inside OSv:
+```
+# content of file /osv-runtime/burek:
+node burek.js
+```
+
+When user runs `capstan package compose` the boot command is set so that the agent is run. A capstan package must be implemented that provides agent funcionality.
+
+#### OSv Agent
+Agent reads *cloud-init* file. Based on cloud-init data it spawns a thread with appropriate run command.
```

Comment:
The user provides a file, but this needs to be served somehow. Check https://cloudinit.readthedocs.io/en/latest/topics/datasources.html for various options. OpenStack uses EC2 mechanism, namely it provides a "dummy" IP and service from where the guest OS can get the information about the user. In OSv this works as well when executed in OpenStack.

However, if we are to support a local environment using the same cloud-init approach, it needs to get the data somehow. Since Captan invokes QEMU, the only viable option is to attach a second volume to it. Basically, the main question is actually how are we going to support multiple runtimes in local execution?

Will Capstan set the command line appropriately for us? For example `capstan run --runtime=burek` is going to set the command line of the instance to `agent run burek` or something like that?

Finally, the reason why this works is because the commit I referenced above looks for a particular device file and reads raw data from it. So, this is not entirely a disk volume as you know it from OpenStack, but rather just a raw file that is read by the cloud-init module.